### ISSUE TITLE
NCCL version appears broken for minor release number >= 10

### ISSUE THREAD

**Issue by eqy** [NVIDIA/nccl#543]
https://github.com/NVIDIA/nccl/blob/7e515921295adaab72adf56ea71a0fafb0ecb5f3/src/nccl.h.in#L22

It appears only a single digit is allocated for the minor release number, so e.g., 2.10.x rolls over into 3.0.x. 

Linked issue in PyTorch: https://github.com/pytorch/pytorch/issues/62295
As it happens, 2.10.3 is a relatively important release as it implements `bfloat16` support.

**Comment by AddyLaddy** [comment]
As the code above shows, we extended the NCCL minor number to be two digits in the 2.9.x releases and later to resolve that issue.
The PyTorch code has hard coded the previous NCCL version number encoding and so reports the version incorrectly in the error messages.

**Comment by eqy** [comment]
I'm a bit confused by this change. It looks like 2.10.3 would be changed to 20103 in this case, but then 3.0.0 would revert to 3000. Is this the intent or am I mistranslating this? It seems that this would break simple "greater than" comparisons.

**Comment by AddyLaddy** [comment]
Yeah sorry you're correct. I was trying to make sure we used the old encoding for versions before 2.9.x.
It looks like I need to change that to be

 `#define NCCL_VERSION(X,Y,Z) (((X) >= (3) || ((X) >= 2 && (Y) >= 9)) ? (X) * 10000 + (Y) * 100 + (Z) : (X) * 1000 + (Y) * 100 + (Z))
`
Or drop the conditional altogether. We'll fix it in the next release. 
We don't have any plans for a NCCL 3.x.x just yet, so it will work fine and generate increasing values for all versions up to 2.99.99

**Comment by stephenmsachs** [comment]
### Issue Cleanup: Helping Us Focus on Current Challenges

  We're reviewing older issues to ensure we prioritize the most relevant and active ones. Since this issue hasn't seen updates in over 6 months, we'll be closing it for now.

  ,*This change helps us focus our efforts on addressing any current issues our users are facing.* If this issue still affects you, please don't hesitate to reopen it with a quick update (e.g., "Still relevant on [version=X]").
  Thanks for your understanding and for contributing to NCCL.
